Firestore/core/include/firebase/firestore/firestore.h

```diff
@@ -0,0 +1,162 @@
+
+// TODO(rsgowman): replace these forward decl's with appropriate includes (once
+// they exist)
+class DocumentReference;
+class CollectionReference;
+class Settings;
+
+/**
+ * @brief Entry point for the Firebase Firestore C++ SDK.
+ *
+ * To use the SDK, call firebase::firestore::Firestore::GetInstance() to obtain
+ * an instance of Firestore, then use Collection() or Document() to obtain
+ * references to child paths within the database. From there you can set data
+ * via CollectionReference::Add() and DocumentReference::Set(), or get data via
+ * CollectionReference::Get() and DocumentReference::Get(), attach listeners,
+ * and more.
+ *
+ * Subclassing Note: Firestore classes are not meant to be subclassed except for
+ * use in test mocks. Subclassing is not supported in production code and new
+ * SDK releases may break code that does so.
+ */
+class Firestore {
+ public:
+  /**
+   * @brief Get an instance of Firestore corresponding to the given App.
+   *
+   * Firebase Firestore uses firebase::App to communicate with Firebase
+   * Authentication to authenticate users to the Firestore server backend.
+   *
+   * If you call GetInstance() multiple times with the same App, you will get
+   * the same instance of App.
+   *
+   * @param[in] app Your instance of firebase::App. Firebase Firestore will use
+   * this to communicate with Firebase Authentication.
+   * @param[out] init_result_out Optional: If provided, write the init result
+   * here. Will be set to kInitResultSuccess if initialization succeeded, or
+   * kInitResultFailedMissingDependency on Android if Google Play services is
+   * not available on the current device.
+   *
+   * @returns An instance of Firestore corresponding to the given App.
+   */
+  static Firestore* GetInstance(::firebase::App* app,
+                                InitResult* init_result_out = nullptr);
+
+  static Firestore* GetInstance(InitResult* init_result_out = nullptr);
+
+  /**
+   * @brief Destructor for the Firestore object.
+   *
+   * When deleted, this instance will be removed from the cache of Firestore
+   * objects. If you call GetInstance() in the future with the same App, a new
+   * Firestore instance will be created.
+   */
+  virtual ~Firestore();
+
+  /**
+   * @brief Get the firebase::App that this Firestore was created with.
+   *
+   * @returns The firebase::App this Firestore was created with.
+   */
+  virtual const App* app() const;
+
+  /**
+   * @brief Get the firebase::App that this Firestore was created with.
+   *
+   * @returns The firebase::App this Firestore was created with.
+   */
+  virtual App* app();
+
+  /**
+   * @brief Gets a CollectionReference instance that refers to the collection at
+   * the specified path within the database.
+   *
+   * @param[in] collectionPath A slash-separated path to a collection.
+   *
+   * @return The CollectionReference instance.
+   */
+  virtual CollectionReference Collection(const char* collectionPath) const;
```

Comment:
Sorry I just noticed this: I think style guide presumes snake_case for parameter names (here and elsewhere).

Reply:
Done; throughout.